proto/osmosis/incentives/gauge.proto

```diff
@@ -41,4 +41,45 @@ message LockableDurationsInfo {
     (gogoproto.stdduration) = true,
     (gogoproto.moretags) = "yaml:\"lockable_durations\""
   ];
+}
+
+message HistoricalReward {
+  // Reward per share for each denom
+  repeated cosmos.base.v1beta1.DecCoin cumulative_reward_ratio = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.DecCoins"
+  ];
+  uint64 period = 2;
+  uint64 last_eligible_epoch = 3;
+}
+
+message CurrentReward {
+  uint64 period = 1;
+  int64 last_processed_epoch = 2;
+
+  // total coin locked per denom + lockDuration
+  cosmos.base.v1beta1.Coin coin = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coin"
+  ];
+  repeated cosmos.base.v1beta1.Coin rewards = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+  string denom = 5;
+  google.protobuf.Duration lock_duration = 6 [
+    (gogoproto.stdduration) = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"duration\""
+  ];
+}
+
+message PeriodLockReward {
```

Comment:
I don't understand just via reading the proto files what this struct is for.

Is it the return value when getting rewards for a lock?

Reply:
Correct, we store how much rewards each individual locks are eligible for.

Reply:
if its per lock, why is it `repeated LastEligibleEpochByDurationAndDenom last_eligible_epochs = 3;` instead of just a single entry for `last_claimed_epoch`

Reply:
Same answer as in the comment below